src/components/ui/SortableList/index.tsx

```diff
@@ -0,0 +1,269 @@
+        showDragOverlay,
+        ...otherProps
+    } = props;
+    const [activeId, setActiveId] = useState<string | undefined>();
+
+    const sensors = useSensors(
+        useSensor(PointerSensor),
+        useSensor(KeyboardSensor, {
+            coordinateGetter: sortableKeyboardCoordinates,
+        }),
+    );
+
+    // NOTE: Sortable context requires list of items
+    const items = useMemo(() => (
+        data?.map(d => String(keySelector(d)))
+    ), [data, keySelector]);
+
+    const handleDragStart = useCallback((event: DragStartEvent) => {
+        const { active } = event;
+        setActiveId(active.id);
+    }, []);
+
+    const handleDragEnd = useCallback((event: DragEndEvent) => {
+        const { active, over } = event;
+        setActiveId(undefined);
+
+        if (active.id && over?.id && active.id !== over?.id && items) {
+            const oldIndex = items.indexOf(active.id);
+            const newIndex = items.indexOf(over.id);
+
+            const newItems = arrayMove(items, oldIndex, newIndex);
+            const dataMap = listToMap(
+                data,
+                d => String(keySelector(d)),
+                d => d,
+            );
+            const newData = newItems.map(item => dataMap[item]);
+            onChange(newData, name);
+        }
+    }, [keySelector, items, data, onChange, name]);
+
+    const DragItem = useMemo(() => {
+        if (!activeId || !data || !showDragOverlay) {
+            return null;
+        }
+        const activeIndex = data.findIndex(
+            d => String(keySelector(d)) === activeId,
+        );
+        if (!activeIndex) {
+            return null;
+        }
+        const params = rendererParams(
+            keySelector(data[activeIndex]),
+            data[activeIndex],
+            activeIndex,
+            data,
+        );
+        return (
+            <Renderer
+                {...params}
+            />
+        );
+    }, [
+        activeId,
+        Renderer,
+        keySelector,
+        rendererParams,
+        data,
+        showDragOverlay,
+    ]);
+
+    const modifiedRendererParams = useCallback((
+        _: K,
+        datum: D,
+        index: number,
+        dataFromArgs: D[],
+    ) => {
+        const params = rendererParams(
+            keySelector(datum),
+            datum,
+            index,
+            dataFromArgs,
+        );
+
+        return ({
+            params,
+            datum,
+            keySelector,
+            renderer: Renderer,
+        });
+    }, [keySelector, rendererParams, Renderer]);
+
+    return (
+        <DndContext
+            sensors={sensors}
+            collisionDetection={closestCenter}
+            onDragStart={handleDragStart}
+            onDragEnd={handleDragEnd}
+            modifiers={[
+                direction === 'horizontal' ? restrictToHorizontalAxis : restrictToVerticalAxis,
+            ]}
+        >
+            <SortableContext
+                items={items ?? []}
```

Comment:
```suggestion
                items={items ?? []}
```
Let's not create new elements on render